Changes so variables starting with tru / fal will no longer fail

cwisniew opened the pull request:
Its not a 100% fix, for example 
falseSenseOfSecurity = 1 
will still fail... but it's still an improvement over things like falling = 1 failing...

partially fixes RPTools/MapTool#1736

A proper fix would require not tokenising "true"/"false" and having them as parser rules which can make decisions based on context rather than lexer which can not. But this is too big a change at this point in time.

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/rptools/parser/50)
<!-- Reviewable:end -->

cwisniew committed:
Changes so variables starting with tru / fal will no longer fail

Merudo reviewed (commented):
> Reviewed 1 of 1 files at r1.
> *[Reviewable](https://reviewable.io/reviews/rptools/parser/50)* status: :shipit: complete! all files reviewed, all discussions resolved



<!-- Sent from Reviewable.io -->

cwisniew committed:
TRUE/FALSE are no longer tokens but constants

cwisniew committed:
no longer need 5 lookahead for parser

cwisniew commented:
Updated so that true/false are no longer tokens and will instead be identifiers. Added code to MapVariableResolver to treat these as constant (as its the closest thing that this parser has to a symbol table). Similar code will need to be added to MapToolVariableResolver (and has been done as part of RPTools/maptool#2168)

Phergus reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

> Reviewed 6 of 6 files at r2.
> *[Reviewable](https://reviewable.io/reviews/rptools/parser/50#-:-MErmzBRCNVlC3sfFC_I:bnfp4nl)* status: :shipit: complete! all files reviewed, all discussions resolved



<!-- Sent from Reviewable.io -->
